.github/workflows/lint.yml

```diff
@@ -38,7 +38,7 @@ jobs:
         uses: "shivammathur/setup-php@v2"
         with:
           coverage: "none"
-          php-version: "${{ matrix.php-version }}"
+          php-version: "8.4"
```

Comment:
I'd do this a bit differently:

1) Install `${{ matrix.php-version }}`
2) Run the composer steps
3) Only when downgrading: install 8.4
4) Transform source code
5) Only when downgrading: install `${{ matrix.php-version }}` again

You can try playing with https://docs.github.com/en/actions/tutorials/create-actions/create-a-composite-action to see if we can achieve DRY across different workflows.

Reply:
What I want to avoid is always start by installing 8.4 which is immediately going to be ovewritten with matrix version.

Reply:
> You can try playing with [docs.github.com/en/actions/tutorials/create-actions/create-a-composite-action](https://docs.github.com/en/actions/tutorials/create-actions/create-a-composite-action) to see if we can achieve DRY across different workflows.

I would do that in a followup

Reply:
not sure what todo with the "Compile PHAR" - or what the actual problem is.

Reply:
Can you link the failure? I don't see anything big.

If you mean "Checksum PHAR" - that's fine. The idea behind that job is that the PHAR checksum should not change when src/ doesn't change.

If we change something about the compilation process then it's fine if PHAR contents change. I suspect that some ConstFetches now use FQN like `\null` so the source slightly changed.

Reply:
> If you mean "Checksum PHAR" - that's fine.

yep, that one. wasn't sure what it means